pkg/sql/opt_exec_factory.go

```diff
@@ -1629,7 +1629,7 @@ func (ef *execFactory) ConstructDelete(
 
 	fastPathInterleaved := canDeleteFastInterleaved(tabDesc, fkTables)
 	if fastPathNode, ok := maybeCreateDeleteFastNode(
-		context.TODO(), input.(planNode), tabDesc, fastPathInterleaved, rowsNeeded); ok {
+		context.TODO(), input.(planNode), tabDesc, fkTables, fastPathInterleaved, rowsNeeded); ok {
```

Comment:
Can you add a comment that `canDeleteFastInterleaved` actually modifies `fkTables` and adds all of the interleaved children? It wasn't clear to me that this was the case and I was confused why `fkTables` was the correct thing to pass in.

Reply:
can you update this to not use `context.TODO()` and instead use the context at the top of the function?

Reply:
removed `context.TODO()`.

Looking at `canDeleteFastInterleaved()`, I'm not sure it modifies `fkTables`. We iterate over `fkTables`, for sure, but where does it modify?

Reply:
Sorry, I read the code wrong. I thought the loop that `canDeleteFastInterleaved()` was modifying fkTables. 

After a second glance, I see why this is correct. By default, `fkTables` doesn't hold all of the interleaved children of the table. However, for us to actually enter the fast path for deletion, all the interleaved children have to be in `fkTables`, so it is correct for `fkTables` to be used as the set of tables given to the fetcher. Does my reasoning match what you intended?